### DISCUSSION TITLE
Query self context

### DISCUSSION THREAD

**Discussion by ghost** [mikro-orm/mikro-orm#2192] [Q&A] [answered]
Hey,
When writing my `findDiscardable()`  function for my infraction entity's repository, I want to query all the rows whose `duration` field is less than or equal to `Date.now()` subtract the issuance timestamp of the infraction.

```ts
function findDiscardable() {
  return this.find({
    penalty: [InfractionPenalty.Suspend, InfractionPenalty.Freeze, InfractionPenalty.Lock],
    duration: { $lte: Date.now() - this.issuedAt }, // Because `findDiscardable()` would be a class method, `this` would refer to the class. I just want to know if there's a way to accomplish what I'm trying to do using `this` as a row in the table. I'm not sure if a query like this is even possible with SQL.
  });
}
````

**Comment by B4nan** [comment, marked as answer]
You will need to use query builder for this:

https://mikro-orm.io/docs/query-builder#referring-to-column-in-update-queries

Using `qb.getResult()` you will get pretty much the same response as from `em.find()` - a list of managed entities.

Something like this should work:

```ts
function findDiscardable() {
  // you will need to extend `EntityRepository` exported from your driver package to have this method
  const qb = this.createQueryBuilder('e');

  return qb.select('*').where({
    penalty: [InfractionPenalty.Suspend, InfractionPenalty.Freeze, InfractionPenalty.Lock],
    duration: { $lte: qb.raw('now() - e.issued_at') },
  }).getResult();
}
```

**Reply by ghost** [reply]
Ah alright, thank you.
I would've preferred to keep the ORM methods (e.g., the `find()` method) for consistency throughout my code, but if I can't, the query builder is fine. Are there any plans to support row context within the ORM methods in MikroORM soon?

**Reply by B4nan** [reply]
Actually it should even work with `em.find()`, but you need to create the QB to have access to `qb.raw()`. You would also have to escape strictly typed `FilterQuery` via `expr()` helper. So overall better to go with QB directly, not everything will be supported on EM level.
